docs/tutorials/event-driven-microservice.md

```diff
@@ -0,0 +1,613 @@
+        Mail mail = new Mail(from, subject, to, content);
+
+        Request request = new Request();
+        try {
+            request.setMethod(Method.POST);
+            request.setEndpoint("mail/send");
+            request.setBody(mail.build());
+            Response response = sg.api(request);
+            System.out.println("Attempted to send email!\n");
+            System.out.println("Status code: " + response.getStatusCode());
+            System.out.println("Body: " + response.getBody());
+            System.out.println("Headers: " + response.getHeaders());
+            System.out.println("======================");
+        } catch (IOException ex) {
+            throw ex;
+        }
+    }
+
+    private static String makeSubject(PossibleAnomaly anomaly) {
+        return "Suspicious activity detected for card " + anomaly.getCardNumber();
+    }
+
+    private static String makeContent(PossibleAnomaly anomaly) {
+        return String.format("Found suspicious activity for card number %s. %s transactions were made for a total of %s between %s and %s",
+                anomaly.getCardNumber(),
+                anomaly.getNAttempts(),
+                anomaly.getTotalAmount(),
+                formatter.format(Instant.ofEpochMilli(anomaly.getStartBoundary())),
+                formatter.format(Instant.ofEpochMilli(anomaly.getEndBoundary())));
+    }
+
+}
+```
+
+### Run the microservice
+
+Compile the program with:
+
+```
+mvn compile
+```
+
+And run it:
+
+```
+mvn exec:java -Dexec.mainClass="io.ksqldb.tutorial.EmailSender"
+```
+
+If everything is configured correctly, emails will be sent whenever an anomaly is detected. There are a few things to note with this simple implementation.
+
+First, if you start more instances of this microservice, the partitions of the `possible_anomalies` topic will be load balanced across them. This takes advantage of the standard [Kafka consumer groups](https://kafka.apache.org/documentation/#intro_consumers) behavior.
+
+Second, this microservice is configured to checkpoint its progress every `100` milliseconds through the `ENABLE_AUTO_COMMIT_CONFIG` configuration. That means any successfully processed messages will not be reprocessed if the microservice is taken down and turned on again.
+
+Finally, note that ksqlDB emits a new event every time a tumbling window changes. ksqlDB uses a model called refinements to continutally emit new changes to stateful aggregations. For example, if an anomaly was detected because 3 credit card transactions were found in a given interval, an event would be emitted from the table. If a 4th was detected in the same interval, another event will be emitted. Because SendGrid does not (at the time of writing) support idempotent email submission, you would need to have a small piece of logic in your program to prevent sending an email multiple times for the same period. We've ommitted this for brevity.
```

Comment:
```suggestion
Finally, note that ksqlDB emits a new event every time a tumbling window changes. ksqlDB uses a model called "refinements" to continually emit new changes to stateful aggregations. For example, if an anomaly was detected because three credit card transactions were found in a given interval, an event would be emitted from the table. If a fourth is detected in the same interval, another event is emitted. Because SendGrid does not (at the time of writing) support idempotent email submission, you would need to have a small piece of logic in your program to prevent sending an email multiple times for the same period. This is omitted for brevity.
```